src/ResourceManager/Compute/Commands.Compute/Extension/Chef/SetAzureVMChefExtension.cs

```diff
@@ -263,38 +271,86 @@ private Hashtable PublicConfiguration
                     {
```

Comment:
Change the order of conditions to make the code simpler.  Put if(IsRunListEmpty) condition inside of other condition, and you can avoid repetition.